api/v1alpha1/hostedcluster_types.go

```diff
@@ -70,109 +70,151 @@ const (
 	MachineApproverImage = "hypershift.openshift.io/machine-approver-image"
 )
 
-// HostedClusterSpec defines the desired state of HostedCluster
+// HostedClusterSpec is the desired behavior of a HostedCluster.
 type HostedClusterSpec struct {
-
-	// Release specifies the release image to use for this HostedCluster
+	// Release specifies the OCP release used for the control plane.
 	Release Release `json:"release"`
 
-	// +optional
+	// InfraID is a globally unique identifier for the cluster. This identifier
+	// will be used to associate various resources with the HostedCluster and its
+	// associated NodePools.
+	//
 	// +immutable
-	FIPS bool `json:"fips"`
+	InfraID string `json:"infraID"`
 
-	// PullSecret is a pull secret injected into the container runtime of guest
-	// workers. It should have an ".dockerconfigjson" key containing the pull secret JSON.
+	// Platform specifies the underlying infrastructure provider for the cluster
+	// and is used to configure platform specific behavior.
+	//
 	// +immutable
-	PullSecret corev1.LocalObjectReference `json:"pullSecret"`
+	Platform PlatformSpec `json:"platform"`
 
-	// AuditWebhook contains metadata for configuring an audit webhook
-	// endpoint for a cluster to process cluster audit events. It references
-	// a secret that contains the webhook information for the audit webhook endpoint.
-	// It is a secret because if the endpoint has MTLS the kubeconfig will contain client
-	// keys. This is currently only supported in IBM Cloud. The kubeconfig needs to be stored
-	// in the secret with a secret key name that corresponds to the constant AuditWebhookKubeconfigKey.
+	// ControllerAvailabilityPolicy specifies the availability policy applied to
+	// critical control plane components. The default value is SingleReplica.
```

Comment:
should the comment say what the possible values are?

Reply:
I've also thought about that...

If you click the `AvailabilityPolicy` link for the field, you can see the enum values in their own section. I've also experimented with in-lining the info from the enum description table into the comment for the field like you're suggesting, but with the current probably overly narrow field for the description it looked like too much to me (IMO) so I left it out for now thinking a little more work is required to make better use of space generally. I do like the idea of putting some or all of the enum info right inline one way or another as a quick reference.

The good news is we have that context in the template data (e.g. indicator of a field being an enum, functions to list the enum values and descriptions, etc), so what do you think about addressing the structural change in a separate PR which would improve this uniformly for all enum fields?

Reply:
sgtm

Reply:
@csrwng this was bugging me so I spent some time on it and here's what I came up with:

1. I enabled the [integrated TOC feature](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#integrated-table-of-contents) of the Material theme so that the TOC is now integrated into the left-side navigation bar instead of being its own section on the right, which IMO is nicer looking and also frees up like 30% of the space on the right side of the display, which makes the API contents wider and hopefully easier to read
2. I tried a middle ground approach to dealing with enums... whenever a field is an enum, you now get the description of the field with an additional note below the description like "Value must be one of: x, y, z" listing the valid constant values, and the user can click the field type to go to the constant and see detailed info about the enum.

Check out the latest preview and let me know what you think...